src/qmxgraph/page/api.js

```diff
@@ -1709,8 +1709,14 @@ graphs.Api.prototype.setEdgeTerminal = function setEdgeTerminal (
         }
         var edge_style = model.getStyle(edge);
         var terminal_key = terminalType + 'Port';
-        console.log('setEdgeTerminal', 'edge_style', edge_style, terminal_key, terminal_value);
-        edge_style = graphs.utils.setStyleKey(edge_style, terminal_key, terminal_value);
+        if (portName) {
+            edge_style = graphs.utils.setStyleKey(edge_style, terminal_key, terminal_value);
+        }
+        else {
+            // Remove the previously connected port (tries both options).
+            edge_style = graphs.utils.removeStyleKey(edge_style, graphs.Api.SOURCE_TERMINAL_CELL + 'Port');
+            edge_style = graphs.utils.removeStyleKey(edge_style, graphs.Api.TARGET_TERMINAL_CELL + 'Port');
```

Comment:
This `edge_style` will override the previous, seems strange given we use `edge_style` later on.

Reply:
This was intended, I don't know which port was set before, so I try to remove both.

Reply:
Nah...
You should remove the one respective to the terminal being changed. The "key" is already in `terminal_key`.

Reply:
You are right, will update it.